nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/hazelcast/HazelcastCore.java

```diff
@@ -116,6 +118,8 @@
     private String memberName;
     private String memberGroup;
 
+    private boolean datagridEncryptionChanged = false;
```

Comment:
I wouldn't say it's any more of an issue than the other variables - changes are made transactionally.

Reply:
If the `@Service` does not cause some wrapper or extension of the class that does some black magic then I don't see how the field is properly made visible when one thread is reading while another one is writing. This might be my lack of knowledge on HK2 or it is actually a problem. I sure looks like one to me. I'd consider the possibility that the same problem exists for the other values :D